plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/semantics/model/ddl/SQLQueryTableDropModel.java

```diff
@@ -94,7 +94,13 @@ public SQLQueryDataContext getGivenDataContext() {
     @Override
     protected void applyContext(@NotNull SQLQueryDataContext dataContext, @NotNull SQLQueryRecognitionContext recognitionContext) {
         this.dataContext = dataContext;
+        if (ifExists) {
+            recognitionContext.setTreatErrorAsWarnings(true);
```

Comment:
I wonder if there might be a situation when this can overwrite the flag at an appropriate time. What I mean by that is that if the caller explicitly sets to `true`, let's say, it will be set to `false` when this method exists, and it might not be the desired behavior.

So I was thinking, what if you make it a stack so you can push and pop different states?

Reply:
I don't see how such a situation may happen

Reply:
Yes, potentially it could be implemented like a stack